duplicacy/duplicacy_main.go

```diff
@@ -687,14 +689,21 @@ func restoreRepository(context *cli.Context) {
             pattern = pattern[1:]
         }
 
-        if pattern[0] != '+' && pattern[0] != '-' {
+        if pattern[0] != '+' && pattern[0] != '-' && pattern[0] != 'i' && pattern[0] != 'e' {
             pattern = "+" + pattern
         }
 
-        if pattern == "+" || pattern == "-" {
+        if pattern == "+" || pattern == "-" || pattern == "i:" || pattern == "e:" {
             continue
         }
 
+        if strings.HasPrefix(pattern, "i:") || strings.HasPrefix(pattern, "e:") {
+            _, err := regexp.Compile(pattern[2:])
+            if err != nil {
+                duplicacy.LOG_ERROR("SNAPSHOT_FILTER", "Invalid regular express encountered for filter: \"%s\", error: %v", pattern, err)
```

Comment:
Is logging the error sufficient here? Do we really want to just ignore the invalid regex and proceed as though everything else was the sum total of what we were asked to do?

I think we'd want to just stop the operation outright, to avoid doing something invalid/incorrect (which might cost the user money, and/or might give them a false sense that everything was working when it wasn't).

Reply:
In my testing, this does stop the operation after the invalid regex. Are you finding that not to be the case?

Reply:
My comments were based purely on reading the code not running it, and I'm still coming up to speed on Go, so I very well could have misinterpreted something. I would have expected there to be a return after the LOG_ERROR call, which was what I was commenting based on, but I haven't attempted to actually test the code by running it.

Reply:
I don't know go either, but I definitely confirmed that LOG_ERROR will stop the program after logging the message.